101-bundle-json.md

```diff
@@ -442,22 +442,24 @@ Implementations MAY support user-defined additional actions as well. Such action
 
 ```json
 "actions": {
-    "status":{
+    "io.cnab.status":{
         "modifies": false,
         "description": "retrieves the status of an installation"
     },
-    "migrate":{
+    "io.cnab.migrate":{
         "modifies": false
     },
-    "dry-run":{
+    "io.cnab.dry-run":{
         "modifies": false,
         "stateless": true,
         "description": "prints what install would do with the given parameters values"
     }
 }
 ```
 
-The above declares to actions: `status` and `migrate`. This means that the associated invocation images can handle requests for `status` and `migrate` in addition to `install`, `upgrade`, and `uninstall`.
+The action _name_ SHOULD be namespaced and SHOULD use reverse domains notation - e.g. `com.example.myKey`. 
+
+The above declares three actions: `io.cnab.status`, `io.cnab.migrate` and `io.cnab.dry-run`. This means that the associated invocation images can handle requests for `status`, `migrate` and `dry-run` in addition to `install`, `upgrade`, and `uninstall`.
```

Comment:
If invocation images are passed truncated action names (e.g. `status`) rather than fully-qualified action names (e.g. `io.cnab.status`), there could be collisions. Why not pass the fully-qualified action names to invocation images instead?

Reply:
And we should at least be specific about what would happen if the action was named `io.cnab.this.is.a.long.name`: is it just the last string (`name`) or `this.is.a.long.name`?

Reply:
Why do you want to treat the prefix `io.cnab` specially? Is it because you'd like standard action names to have fully-qualified synonyms? For example, would you like `io.cnab.install` to be a synonym for `install`?

I don't really think this is necessary because actions with a single component, e.g. `install`, are clearly distinct from reverse domain names. (However, it would be a bad idea to name a standard action after a top-level domain name such as `org` as that would then also be a reverse domain name!)

So, if you agree that we can use single component names for standard actions, then we can treat the above example as a regular custom name: `io.cnab.this.is.a.long.name` would not have a short form. In other words, we wouldn't need to treat the prefix `io.cnab` specially.

Reply:
+1 on @glyn point. I don't think the special casing is worth the extra effort in either the spec to describe the edge cases or in making all clients more complicated to deal with that logic.

I would remove this paragraph.

I think the rationale for doing so would be user interface, but that's a client concern, not for the spec.

Reply:
I'm fine with passing fully-qualified action names, any objection @technosophos @garethr ?